app/views/works/_work_header_navigation.html.erb

```diff
@@ -51,10 +51,15 @@
       <%= link_to ts("Cancel Bookmark"), "#", :class => "hidden bookmark_form_placement_close" %>
     </li>
 
-    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false || marked_for_later?(@work) %>
-      <li class="mark"><%= marktoread_link(@work) %></li>
+    <% unless current_user.is_author_of?(@work) || current_user.try(:preference).try(:history_enabled) == false %>
+     <li class="mark"> 
+      <% if marked_for_later?(@work) %>
```

Comment:
This patch moves mark to read later/mark as read to an immediate write to the database.